buildSrc/src/main/kotlin/io.opentelemetry.instrumentation.javaagent-instrumentation.gradle.kts

```diff
@@ -1,12 +1,21 @@
 plugins {
   id("io.opentelemetry.instrumentation.javaagent-testing")
   id("io.opentelemetry.instrumentation.muzzle-check")
-  id("io.opentelemetry.instrumentation.javaagent-codegen")
+  id("io.opentelemetry.instrumentation.muzzle-generation")
 }
 
 dependencies {
   add("muzzleBootstrap", "io.opentelemetry.instrumentation:opentelemetry-instrumentation-api-annotation-support")
   add("muzzleBootstrap", "io.opentelemetry.javaagent:opentelemetry-javaagent-instrumentation-api")
   add("muzzleTooling", "io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api")
   add("muzzleTooling", "io.opentelemetry.javaagent:opentelemetry-javaagent-tooling")
+  add("muzzleTooling", "org.slf4j:slf4j-simple")
+
+  /*
+    Dependencies added to this configuration will be found by the muzzle gradle plugin during code
+    generation phase. They differ from the implementation dependencies declared in plugin's build
+    script, because these classes become part of the code that plugin inspects and traverses during
+    references collection phase. They are not part of the observer, they are part of the observation.
```

Comment:
Is javaagent-extension-api needed in the buildSrc dependencies? It seems like something the observer wouldn't need, only the observation.

Reply:
Are you talking about `muzzleTooling` dependency above? That is for muzzle checking, I have not touched it yet.

Reply:
Sorry I didn't mean buildSrc, but gradle-plugins (it feels like buildSrc :P) Is this one needed?

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/gradle-plugins/build.gradle.kts#L25

Reply:
Yes, `MuzzleCodeGenerator` uses it:

```
import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
```

Take a look at this method's content: https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/85b7c0569e7fa10aaa97ca9acc07f830753dfb0f/gradle-plugins/src/main/java/io/opentelemetry/javaagent/muzzle/generation/MuzzleCodeGenerator.java#L161